### PR TITLE
[SYCL] Add tests for atomic_fence_capabilities.

### PR DESCRIPTION
**Pull request by maarquitos14**
Adds E2E tests for `sycl::info::device::atomic_fence_order_capabilities` and `sycl::info::device::atomic_fence_scope_capabilities`.

### CODE REVIEW

**Review by steffenlarsen — APPROVED** [review]
Looks good!

File: sycl/test-e2e/AtomicRef/atomic_fence_capabilities.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+                        memory_order mem_order) {
+  return std::find(capabilities.begin(), capabilities.end(), mem_order) !=
+         capabilities.end();
+}
+
+bool is_supported_scope(const std::vector<memory_scope> &capabilities,
+                        memory_scope mem_scope) {
+  return std::find(capabilities.begin(), capabilities.end(), mem_scope) !=
+         capabilities.end();
+}
```
Comment (bader) [inline sycl/test-e2e/AtomicRef/atomic_fence_capabilities.h:18]: Please, move this code to the cpp file. Having function definitions in a header file is fraught with problems.

**Comment by bader** [comment]
@maarquitos14, sorry, I merged this PR by mistake.
Please, address fix the failure on CUDA backend and create a new PR.

**Comment by maarquitos14** [comment]
> @maarquitos14, sorry, I merged this PR by mistake. Please, address fix the failure on CUDA backend and create a new PR.

The [PR](https://github.com/intel/llvm/pull/8901) is already created, I'm waiting for review.

**Pull request merged**